server/src/main/java/org/elasticsearch/common/lucene/search/function/ScriptScoreFunction.java

```diff
@@ -62,12 +74,25 @@ public LeafScoreFunction getLeafScoreFunction(LeafReaderContext ctx) throws IOEx
         leafScript._setIndexName(indexName);
         leafScript._setShard(shardId);
         return new LeafScoreFunction() {
+            private String customExplanation = null;
+
             @Override
             public double score(int docId, float subQueryScore) throws IOException {
                 leafScript.setDocument(docId);
                 scorer.docid = docId;
                 scorer.score = subQueryScore;
-                double result = leafScript.execute(null);
+                double result;
```

Comment:
I don't know why we are updating the `score` here at all. Doesn't this work with `explainScore` ? And if `explainScore` is being called, we know we are "explaining" and thus should create a `ScoreScript.ExplanationHolder` ?

Reply:
That makes a lot of sense; I appreciate all your comments / direction.  I'll come back with changes around all of your comments.

Reply:
I believe all of your concerns have been addressed in the lastest push; let me know if you have additional concerns at this point @benwtrent